### ISSUE TITLE
Prometheus Reader failure

### ISSUE THREAD

**Issue by ekrichbaum** [FlxPeters/netbox-plugin-prometheus-sd#96]
Before I start, thank you for this work and I fully admit this could be me.  No experience with Prometheus at all here.
prometheus-2.42.0.linux-amd64
netbox 3.4.5

API seems to return good data, validly formatted. Ran output through json lint and looks valid.  All elements are strings excep for the id.

```
          "__meta_netbox_id": 81,
```

Prometheus never shows the target at all and logs this error:
```
Feb 23 08:35:13 prometheus.netfortris.com prometheus[11055]: ts=2023-02-23T16:35:13.317Z caller=refresh.go:99 level=error component="discovery manager scrape" discovery=http config=http_sd_devices msg="Unable to refresh target groups" err="json: cannot unmarshal number into Go struct field .labels of type model.LabelValue"
```

Prometheus config:
```
  - job_name: "http_sd_devices"
#    metrics_path: /netbox
    scheme: https
#    static_configs:
#      - targets: ["localhost:9090"]
    http_sd_configs:
    - url: https://netbox.netfortris.net/api/plugins/prometheus-sd/devices/?has_primary_ip=true&name=atl-bdr1
      refresh_interval: 15s
      authorization:
        type: "Token"
        credentials: "blahblahblahforthelength of the token."
```

Validated with curl that the data is returned properly from the url with the proper api key. I get different issues if I try to use the metrics_path so I let it default with the other job into /metrics.

**Comment by cervajs** [comment]
same problem here
we upgraded yesterday from netbox 3.2 (netbox sd plugin - dont remember version) to netbox 3.3 (netbox sd plugin - 0.6)

**Comment by cervajs** [comment]
it looks like a change on netbox side
GET /api/virtualization/virtual-machines/
HTTP 200 OK
Allow: GET, POST, PUT, PATCH, DELETE, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "count": 37,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 43,
            
 so if we want stay prometheus compatible we must change type from integer to string?

**Comment by ekrichbaum** [comment]
```        labels = LabelDict(
            {"status": obj.status, "model": obj.__class__.__name__, "name": obj.name, "id": obj.id}
        )```

would be simple to change to str(obj.id) if that is the case in the serializers.py line 27.  That removes the error.  I still have a use case issue then where it still doesn't do anything properly.  I'll open a different issue for it.

**Comment by FlxPeters** [comment]
Hey, thanks for your input. This seams to be a bug indeed. Cast to string should do the job. I will try to push a fix later if not anybody else is faster with a MR.

**Comment by DonPerinon** [comment]
Just to make it clear there was no change made in netbox. Prior to the .6 version primary keys were not provided as labels. https://github.com/FlxPeters/netbox-plugin-prometheus-sd/blob/11a284021f190b71cedd932a6b245bcd64b6ecdc/netbox_prometheus_sd/api/serializers.py#L2727

**Comment by FlxPeters** [comment]
Fix has been published in 0.6.2

https://github.com/FlxPeters/netbox-plugin-prometheus-sd/releases/tag/0.6.2
